


# P3_Réalisez une étude de santé publique

## Importer  pandas et lire des bases de données

In [38]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [39]:
df_population = pd.read_csv('fr_population.csv')
df_animaux = pd.read_csv('fr_animaux.csv')
df_vegetaux = pd.read_csv('fr_vegetaux.csv')
df_céréales = pd.read_csv('fr_céréales.csv')
df_sousalimentation = pd.read_csv('fr_sousalimentation.csv')

### Connaître la base de données
df_population 
df_animaux 
df_vegetaux 
df_céréales 
df_sousalimentation 

In [40]:
df_population.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole'],
      dtype='object')

In [41]:
df_population.shape

(175, 14)

In [42]:
df_population


Code Domaine                                            Domaine  \
0           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
1           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
2           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
3           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
4           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
..           ...                                                ...   
170         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
171         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
172         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
173         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
174         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

     Code zone                                    Zone  Code Élément  \
0            2                             Afghanistan           511   
1          202                          Afrique du Sud           511   
2            3                                 Albanie           511   
3            4                                 Algérie           511   
4           79                               Allemagne           511   
..         ...                                     ...           ...   
170        236  Venezuela (République bolivarienne du)           511   
171        237                                Viet Nam           511   
172        249                                   Yémen           511   
173        251                                  Zambie           511   
174        181                                Zimbabwe           511   

               Élément  Code Produit     Produit  Code année  Année  \
0    Population totale          2501  Population        2013   2013   
1    Population totale          2501  Population        2013   2013   
2    Population totale          2501  Population        2013   2013   
3    Population totale          2501  Population        2013   2013   
4    Population totale          2501  Population        2013   2013   
..                 ...           ...         ...         ...    ...   
170  Population totale          2501  Population        2013   2013   
171  Population totale          2501  Population        2013   2013   
172  Population totale          2501  Population        2013   2013   
173  Population totale          2501  Population        2013   2013   
174  Population totale          2501  Population        2013   2013   

              Unité  Valeur Symbole Description du Symbole  
0    1000 personnes   30552     NaN      Donnée officielle  
1    1000 personnes   52776     NaN      Donnée officielle  
2    1000 personnes    3173     NaN      Donnée officielle  
3    1000 personnes   39208     NaN      Donnée officielle  
4    1000 personnes   82727     NaN      Donnée officielle  
..              ...     ...     ...                    ...  
170  1000 personnes   30405     NaN      Donnée officielle  
171  1000 personnes   91680     NaN      Donnée officielle  
172  1000 personnes   24407     NaN      Donnée officielle  
173  1000 personnes   14539     NaN      Donnée officielle  
174  1000 personnes   14150     NaN      Donnée officielle  

[175 rows x 14 columns]

In [43]:
df_population.nunique()

Code Domaine                1
Domaine                     1
Code zone                 175
Zone                      175
Code Élément                1
Élément                     1
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       1
Valeur                    175
Symbole                     1
Description du Symbole      2
dtype: int64

In [44]:
df_population.head()

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        202   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          3   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          4   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         79   

             Zone  Code Élément            Élément  Code Produit     Produit  \
0     Afghanistan           511  Population totale          2501  Population   
1  Afrique du Sud           511  Population totale          2501  Population   
2         Albanie           511  Population totale          2501  Population   
3         Algérie           511  Population totale          2501  Population   
4       Allemagne           511  Population totale          2501  Population   

   Code année  Année           Unité  Valeur Symbole Description du Symbole  
0        2013   2013  1000 personnes   30552     NaN      Donnée officielle  
1        2013   2013  1000 personnes   52776     NaN      Donnée officielle  
2        2013   2013  1000 personnes    3173     NaN      Donnée officielle  
3        2013   2013  1000 personnes   39208     NaN      Donnée officielle  
4        2013   2013  1000 personnes   82727     NaN      Donnée officielle

### Question 1 : donnez le résultat de votre calcul pour l'année 2013.

In [45]:
df_population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Code Domaine            175 non-null    object
 1   Domaine                 175 non-null    object
 2   Code zone               175 non-null    int64 
 3   Zone                    175 non-null    object
 4   Code Élément            175 non-null    int64 
 5   Élément                 175 non-null    object
 6   Code Produit            175 non-null    int64 
 7   Produit                 175 non-null    object
 8   Code année              175 non-null    int64 
 9   Année                   175 non-null    int64 
 10  Unité                   175 non-null    object
 11  Valeur                  175 non-null    int64 
 12  Symbole                 1 non-null      object
 13  Description du Symbole  175 non-null    object
dtypes: int64(6), object(8)
memory usage: 19.3+ KB


In [46]:
df_population['Année'] .nunique()

1

Il n'y a que l'année 2013 dans la colonne année.
On peut additionner la population de chaque pays pour obtenir la population totale du monde

In [47]:
df_population .nunique()

Code Domaine                1
Domaine                     1
Code zone                 175
Zone                      175
Code Élément                1
Élément                     1
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       1
Valeur                    175
Symbole                     1
Description du Symbole      2
dtype: int64

In [48]:
df_population ['Description du Symbole']. unique ()

array(['Donnée officielle',
       'Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées'],
      dtype=object)

# population mondiale

In [49]:
population2013 = (df_population['Valeur'].sum()*1000)
population2013

8413993000

In [50]:
print ('La population de la terre est ',population2013 , ', Mais en 2013, la population de la Terre, selon les données des Nations Unies, était de 7 210 581 976 "https://www.worldometers.info/world-population/world-population-by-year/"'
)

La population de la terre est  8413993000 , Mais en 2013, la population de la Terre, selon les données des Nations Unies, était de 7 210 581 976 "https://www.worldometers.info/world-population/world-population-by-year/"


La Chine est comptée deux fois, dans la ligne 33 les données non officielles sont en double
Si nous supprimons) la ligne contenant l'index 33 de notre ensemble de données.

In [51]:
df_population_pas_officielles= (df_population['Description du Symbole']=='Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées')
df_population_pas_officielles

0      False
1      False
2      False
3      False
4      False
       ...  
170    False
171    False
172    False
173    False
174    False
Name: Description du Symbole, Length: 175, dtype: bool

In [82]:
df_population[df_population_pas_officielles]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        NaN   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [85]:

df_population = df_population[df_population['Code zone'] != 40.0]


In [86]:
df_population.iloc[30:40]

Code Domaine                                            Domaine  Code zone  \
30         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       32.0   
31         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       33.0   
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        NaN   
34         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       96.0   
35         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...      128.0   
36         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       41.0   
37         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...      214.0   
38         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       50.0   
39         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       44.0   
40         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...       46.0   

                         Zone  Code Élément            Élément  Code Produit  \
30                   Cameroun           511  Population totale          2501   
31                     Canada           511  Population totale          2501   
33                      Chine           511  Population totale          2501   
34   Chine - RAS de Hong-Kong           511  Population totale          2501   
35       Chine - RAS de Macao           511  Population totale          2501   
36        Chine, continentale           511  Population totale          2501   
37  Chine, Taiwan Province de           511  Population totale          2501   
38                     Chypre           511  Population totale          2501   
39                   Colombie           511  Population totale          2501   
40                      Congo           511  Population totale          2501   

       Produit  Code année  Année           Unité   Valeur Symbole  \
30  Population        2013   2013  1000 personnes    22254     NaN   
31  Population        2013   2013  1000 personnes    35182     NaN   
33  Population        2013   2013  1000 personnes  1416667       A   
34  Population        2013   2013  1000 personnes     7204     NaN   
35  Population        2013   2013  1000 personnes      566     NaN   
36  Population        2013   2013  1000 personnes  1385567     NaN   
37  Population        2013   2013  1000 personnes    23330     NaN   
38  Population        2013   2013  1000 personnes     1141     NaN   
39  Population        2013   2013  1000 personnes    48321     NaN   
40  Population        2013   2013  1000 personnes     4448     NaN   

                               Description du Symbole  
30                                  Donnée officielle  
31                                  Donnée officielle  
33  Agrégat, peut inclure des données officielles,...  
34                                  Donnée officielle  
35                                  Donnée officielle  
36                                  Donnée officielle  
37                                  Donnée officielle  
38                                  Donnée officielle  
39                                  Donnée officielle  
40                                  Donnée officielle

In [ ]:
df_population[]

In [55]:
df_population.shape

(175, 14)

In [56]:
df_population ['Zone']. nunique ()

175

Il y a 206 pays dans le monde. Parmi eux, 193 sont membres des Nations Unies selon https://fr.wikipedia.org/wiki/Organisation_des_Nations_unies_pour_l%27alimentation_et_l%27agriculture, mais dans la base de données actuelle 175 

### Question 2 

Question 2 : Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique (pas besoin de coder ici :soleil: ). C'est une équation à 3 termes de type \(a_1 + a2 + [...] = b_1 + b_2 + [...] = c_1 + c_2 + [...]\) ) faisant intervenir chacune des 11 quantités données ci dessus. Illustrez cette équation avec l'exemple du blé en France. Pour avoir un indice, 

In [58]:
df_vegetaux.head()

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément                   Élément  Code Produit Produit  \
0  Afghanistan          5511                Production          2511     Blé   
1  Afghanistan          5611   Importations - Quantité          2511     Blé   
2  Afghanistan          5072        Variation de stock          2511     Blé   
3  Afghanistan          5301  Disponibilité intérieure          2511     Blé   
4  Afghanistan          5527                  Semences          2511     Blé   

   Code année  Année               Unité  Valeur Symbole  \
0        2013   2013  Milliers de tonnes  5169.0       S   
1        2013   2013  Milliers de tonnes  1173.0       S   
2        2013   2013  Milliers de tonnes  -350.0       S   
3        2013   2013  Milliers de tonnes  5992.0       S   
4        2013   2013  Milliers de tonnes   322.0       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées  
3  Données standardisées  
4  Données standardisées

In [59]:
df_blé_France=((df_vegetaux['Zone']== 'France') & (df_vegetaux['Produit']== 'Blé'))
df_vegetaux[df_blé_France]

Code Domaine                                            Domaine  \
37487         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37488         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37489         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37490         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37491         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37492         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37493         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37494         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37495         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37496         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37497         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37498         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37499         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37500         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37501         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

       Code zone    Zone  Code Élément  \
37487         68  France          5511   
37488         68  France          5611   
37489         68  France          5072   
37490         68  France          5911   
37491         68  France          5301   
37492         68  France          5521   
37493         68  France          5527   
37494         68  France          5123   
37495         68  France          5131   
37496         68  France          5154   
37497         68  France          5142   
37498         68  France           645   
37499         68  France           664   
37500         68  France           674   
37501         68  France           684   

                                                 Élément  Code Produit  \
37487                                         Production          2511   
37488                            Importations - Quantité          2511   
37489                                 Variation de stock          2511   
37490                            Exportations - Quantité          2511   
37491                           Disponibilité intérieure          2511   
37492                              Aliments pour animaux          2511   
37493                                           Semences          2511   
37494                                             Pertes          2511   
37495                                         Traitement          2511   
37496              Autres utilisations (non alimentaire)          2511   
37497                                         Nourriture          2511   
37498  Disponibilité alimentaire en quantité (kg/pers...          2511   
37499     Disponibilité alimentaire (Kcal/personne/jour)          2511   
37500  Disponibilité de protéines en quantité (g/pers...          2511   
37501  Disponibilité de matière grasse en quantité (g...          2511   

      Produit  Code année  Année               Unité    Valeur Symbole  \
37487     Blé        2013   2013  Milliers de tonnes  38614.00       S   
37488     Blé        2013   2013  Milliers de tonnes   2055.00       S   
37489     Blé        2013   2013  Milliers de tonnes   1131.00       S   
37490     Blé        2013   2013  Milliers de tonnes  21502.00       S   
37491     Blé        2013   2013  Milliers de tonnes  20298.00       S   
37492     Blé        2013   2013  Milliers de tonnes   7822.00       S   
37493     Blé        2013   2013  Milliers de tonnes    748.00       S   
37494     Blé        2013   2013  Milliers de tonnes    358.00       S   
37495     Blé        2013   2013  Milliers de tonnes   1575.00       S   
37496     Blé        2013   2013  Milliers de tonnes   2824.00       S   
37497     Blé        2013   2013  Milliers de tonnes   6971.00       S   
37498     Blé        2013   2013                  kg    108.43      Fc   
37499     Blé    

In [60]:
df_vegetaux[df_blé_France].shape

(15, 14)

une équation à 3 termes 

Disponibilité intérieur = Production + importations - exportations + variations des stocks = Nourriture + Aliments pour animaux + Semences + Traitement + Autres utilisations + Pertes